pkg/cloud/services/elb/loadbalancer.go

```diff
@@ -321,6 +331,69 @@ func (s *Service) deleteClassicELBAndWait(name string) error {
 	return nil
 }
 
+func (s *Service) listByTag(tag string) ([]string, error) {
+	input := rgapi.GetResourcesInput{
+		ResourceTypeFilters: aws.StringSlice([]string{elbResourceType}),
+		TagFilters: []*rgapi.TagFilter{
+			{
+				Key:    aws.String(tag),
+				Values: aws.StringSlice([]string{string(infrav1.ResourceLifecycleOwned)}),
+			},
+		},
+	}
+
+	names := []string{}
+
+	err := s.scope.ResourceTagging.GetResourcesPages(&input, func(r *rgapi.GetResourcesOutput, last bool) bool {
+		for _, tagmapping := range r.ResourceTagMappingList {
+			if tagmapping.ResourceARN != nil {
+				// We can't use arn.Parse because the "Resource" is loadbalancer/<name>
+				parts := strings.Split(*tagmapping.ResourceARN, "/")
+				name := parts[len(parts)-1]
+				if name == "" {
+					s.scope.Info("failed to parse ARN", "arn", *tagmapping.ResourceARN, "tag", tag)
+					continue
+				}
+				names = append(names, name)
+			}
+		}
+		return true
+	})
+
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case elb.ErrCodeDependencyThrottleException:
+				return nil, errors.Wrap(err, "too many requests made to the resource tag group service")
```

Comment:
If we don't do this wrapping, what does the error look like in the log? Is it worth the logic to check for this code?

Reply:
Honestly I just did this because `describeClassicELB` did. I'm happy to yank it out.